--- samba-3.4.3.orig/source3/samba4.mk	2009-12-01 16:51:53.810922486 +0100
+++ samba-3.4.3/source3/samba4.mk	2009-12-01 16:52:27.284246504 +0100
@@ -143,6 +143,11 @@
 	@cp $< $@
 	@chmod $(INSTALLPERMS) $@
 
+basics:: regen_version4
+
+regen_version4::
+	@$(SHELL) script/mkversion.sh $(samba4srcdir)/VERSION $(samba4srcdir)/version.h
+
 clean:: 
 	@echo Removing samba 4 objects
 	@-find $(samba4srcdir) -name '*.o' -exec rm -f '{}' \;
