diff -urN talloc-2.0.0.orig/configure.ac talloc-2.0.0/configure.ac
--- talloc-2.0.0.orig/configure.ac	2009-12-01 10:21:43.673256694 +0100
+++ talloc-2.0.0/configure.ac	2009-12-01 10:21:21.683258633 +0100
@@ -36,6 +36,30 @@
 m4_include(libtalloc.m4)
 m4_include(compat/talloc_compat1.m4)
 
+AC_PATH_PROGS([PYTHON_CONFIG], [python2.6-config python2.5-config python2.4-config python-config])
+AC_PATH_PROGS([PYTHON], [python2.6 python2.5 python2.4 python])
+
+PYTHON_BUILD_TARGET="build-python"
+PYTHON_INSTALL_TARGET="install-python"
+PYTHON_CHECK_TARGET="check-python"
+AC_SUBST(PYTHON_BUILD_TARGET)
+AC_SUBST(PYTHON_INSTALL_TARGET)
+AC_SUBST(PYTHON_CHECK_TARGET)
+if test -z "$PYTHON_CONFIG"; then
+        PYTHON_BUILD_TARGET=""
+        PYTHON_INSTALL_TARGET=""
+        PYTHON_CHECK_TARGET=""
+fi
+
+AC_ARG_ENABLE(python,
+                [--enable-python Ebables python bindings],
+                [ if test "x$enableval" = "xno" ; then
+                        PYTHON_BUILD_TARGET=""
+                        PYTHON_INSTALL_TARGET=""
+                        PYTHON_CHECK_TARGET=""
+                  fi
+                ])
+
 DOC_TARGET=""
 AC_ARG_WITH([doc],
 	AS_HELP_STRING([--without-doc], [disable manpage generation]),
diff -urN talloc-2.0.0.orig/Makefile.in talloc-2.0.0/Makefile.in
--- talloc-2.0.0.orig/Makefile.in	2009-12-01 10:21:43.673256694 +0100
+++ talloc-2.0.0/Makefile.in	2009-12-01 10:21:22.436597383 +0100
@@ -23,6 +23,11 @@
 SHLIBEXT = @SHLIBEXT@
 SHLD = @SHLD@
 SHLD_FLAGS = @SHLD_FLAGS@
+PYTHON = @PYTHON@
+PYTHON_CONFIG = @PYTHON_CONFIG@
+PYTHON_BUILD_TARGET = @PYTHON_BUILD_TARGET@
+PYTHON_INSTALL_TARGET = @PYTHON_INSTALL_TARGET@
+PYTHON_CHECK_TARGET = @PYTHON_CHECK_TARGET@
 tallocdir = @tallocdir@
 
 LIBOBJ = $(TALLOC_OBJ) @LIBREPLACEOBJ@
@@ -31,7 +36,7 @@
 VERSIONSCRIPT = @VERSIONSCRIPT@
 EXPORTSFILE = @EXPORTSFILE@
 
-all:: showflags $(EXTRA_TARGETS)
+all:: showflags $(EXTRA_TARGETS) $(PYTHON_BUILD_TARGET)
 
 include $(tallocdir)/rules.mk
 include $(tallocdir)/talloc.mk
@@ -52,6 +57,8 @@
 
 check: test
 
+test:: $(PYTHON_CHECK_TARGET)
+
 installcheck:: test install
 
 distclean:: clean
diff -urN talloc-2.0.0.orig/talloc.mk talloc-2.0.0/talloc.mk
--- talloc-2.0.0.orig/talloc.mk	2009-12-01 10:21:43.676589926 +0100
+++ talloc-2.0.0/talloc.mk	2009-12-01 10:22:02.786589329 +0100
@@ -13,7 +13,13 @@
 	ar -rv $@ $(LIBOBJ)
 	@-ranlib $@
 
-install:: all 
+libtalloc.$(SHLIBEXT): $(TALLOC_SOLIB)
+	ln -fs $< $@
+
+$(TALLOC_SONAME): $(TALLOC_SOLIB)
+	ln -fs $< $@
+
+install:: all $(PYTHON_INSTALL_TARGET)
 	${INSTALLCMD} -d $(DESTDIR)$(libdir)
 	${INSTALLCMD} -d $(DESTDIR)$(libdir)/pkgconfig
 	${INSTALLCMD} -m 755 $(TALLOC_STLIB) $(DESTDIR)$(libdir)
@@ -26,6 +32,10 @@
 	which swig >/dev/null 2>&1 && ${INSTALLCMD} -d $(DESTDIR)`swig -swiglib` || true
 	which swig >/dev/null 2>&1 && ${INSTALLCMD} -m 644 talloc.i $(DESTDIR)`swig -swiglib` || true
 
+install-python:: build-python
+	mkdir -p $(DESTDIR)`$(PYTHON) -c "import distutils.sysconfig; print distutils.sysconfig.get_python_lib(1, prefix='$(prefix)')"`
+	cp talloc.$(SHLIBEXT) $(DESTDIR)`$(PYTHON) -c "import distutils.sysconfig; print distutils.sysconfig.get_python_lib(1, prefix='$(prefix)')"`
+
 doc:: talloc.3 talloc.3.html
 
 clean::
@@ -34,6 +44,14 @@
 	rm -f talloc.exports.sort talloc.exports.check talloc.exports.check.sort
 	rm -f talloc.signatures.sort talloc.signatures.check talloc.signatures.check.sort
 
+build-python:: talloc.$(SHLIBEXT) 
+
+pytalloc.o: $(tallocdir)/pytalloc.c
+	$(CC) $(PICFLAG) -c $(tallocdir)/pytalloc.c $(CFLAGS) `$(PYTHON_CONFIG) --cflags`
+
+talloc.$(SHLIBEXT): libtalloc.$(SHLIBEXT) pytalloc.o
+	$(SHLD) $(SHLD_FLAGS) -o $@ pytalloc.o -L. -ltalloc `$(PYTHON_CONFIG) --ldflags`
+
 test:: testsuite
 	./testsuite
 
@@ -41,6 +59,9 @@
 	@echo ABI checks:
 	@./script/abi_checks.sh talloc talloc.h
 
+check-python:: build-python $(TALLOC_SONAME)
+	$(LIB_PATH_VAR)=. PYTHONPATH=".:$(tallocdir)" $(PYTHON) $(tallocdir)/python/test.py
+
 test:: abi_checks
 
 gcov::
